### DISCUSSION TITLE
是否能添加目录功能

### DISCUSSION THREAD

**Discussion by yingguqing** [tw93/MiaoYan#246] [Ideas]
是否能为文档添加显示[TOC]的功能，方便在文档中查找相应的内容。

**Comment by tw93** [comment]
你是说预览模式下增加toc的效果吗？
这里可以试试锚点的方式，[标题](#标题)
看看能否满足需求

**Comment by vipzhicheng** [comment]
没有目录树管理多层级文件无法从Typora迁移到妙言，其他功能再好也只能放弃。如果坚持单层目录就只能让新用户努力适应了。

**Comment by tw93** [comment]
@vipzhicheng 方便说说多级目录的好处，以及最佳的视觉形式不？以及你一般是几层？

**Comment by wlf92** [comment]
![image](https://user-images.githubusercontent.com/7065616/191176311-1737bb23-bb34-4773-98c5-8e410067c21b.png)
个人感觉目录的好处是为了折叠，可以快速翻到对应内容？

**Comment by vipzhicheng** [comment]
看起来我说的和 @yingguqing 说的是两个东西，层级数应该是不限的，因为笔记结构可能会经常调整。我现在的Typora笔记目录有这样的层级

<img width="247" alt="image" src="https://user-images.githubusercontent.com/1209538/191190665-759514d0-bc81-4a9b-aaa6-66b227f40022.png">

<img width="222" alt="image" src="https://user-images.githubusercontent.com/1209538/191190823-b973e918-c8b9-47b9-bfed-814cf086ee6e.png">

<img width="211" alt="image" src="https://user-images.githubusercontent.com/1209538/191190920-bba0f3a5-a8cd-4113-8e72-ac0b48077e08.png">

如果觉得视觉上树形不好看，可以考虑只显示当前层级，所有的父级都收起来，类似VSCode那样

<img width="805" alt="image" src="https://user-images.githubusercontent.com/1209538/191191718-845dbba3-f53c-4276-8b6a-ee3412f302ca.png">

如果觉得这个层级提示太长，可以只保留root和上一层，中间都用 `...` 代替。

**Comment by zwmmm** [comment]
相比较这种 目录分类 我更加倾向于 熊掌记的标签分类法

**Comment by tw93** [comment]
> 相比较这种 目录分类 我更加倾向于 熊掌记的标签分类法

`+ 1`  标签或许是一种在美观和易用之间的解决方案

**Comment by chunyang-wen** [comment]
> 相比较这种 目录分类 我更加倾向于 熊掌记的标签分类法

熊掌的标签我觉得它的标签顺序好像是有一个排序规则。所以估计还是没有目录结构的清晰稳定。

**Comment by chunyang-wen** [comment]
现在没有目录结构的这个东西，在管理起来相对还是欠缺点。因为就是一个大目录，所以文件都在一起。Ulysses 是有目录结构的，所以管理起来方便点。当然肯定是因人而异。

**Comment by tw93** [comment]
嗯嗯 更像萝卜青菜各有所爱，或者对于妙言本身的特色以及一些“偏执”的地方，会想法子看怎么在产品的层面保持着“特性”，并兼顾一些必备的需求功能，关于目录结构这一块，大伙有比较好的产品形态，不局限于笔记，其他的也行，可以一起看看

**Comment by zwmmm** [comment]
> 现在没有目录结构的这个东西，在管理起来相对还是欠缺点。因为就是一个大目录，所以文件都在一起。Ulysses 是有目录结构的，所以管理起来方便点。当然肯定是因人而异。

有一级目录的啊

**Comment by chunyang-wen** [comment]
> > 现在没有目录结构的这个东西，在管理起来相对还是欠缺点。因为就是一个大目录，所以文件都在一起。Ulysses 是有目录结构的，所以管理起来方便点。当然肯定是因人而异。
> 
> 有一级目录的啊

![](https://user-images.githubusercontent.com/1209538/191190665-759514d0-bc81-4a9b-aaa6-66b227f40022.png)

这位兄台的不行。就是想按照一些比较细节的分类来管理。

嗯嗯，一级目录有。

**Comment by vipzhicheng** [comment]
> > > 现在没有目录结构的这个东西，在管理起来相对还是欠缺点。因为就是一个大目录，所以文件都在一起。Ulysses 是有目录结构的，所以管理起来方便点。当然肯定是因人而异。
> > 
> > 
> > 有一级目录的啊
> 
> ![](https://user-images.githubusercontent.com/1209538/191190665-759514d0-bc81-4a9b-aaa6-66b227f40022.png)
> 
> 这位兄台的不行。就是想按照一些比较细节的分类来管理。
> 
> 嗯嗯，一级目录有。

重点是要灵活，比如我另一个没有展开的目录，是我在某个公司的工作文档，后来换了工作，我就把整个那个公司的文档都移动到一个叫做归档的子目录。意思是文档仍然有价值，但是不常用了。所以我认为目录树这个最基本的文件管理方式是要有的。

**Comment by Tomorrowxxy** [comment]
我也赞成目录树的结构，当前一级目录很不方便。
楼上举例的‘工作’文件夹很贴近使用场景。
不过貌似很多笔记类软件都是一级目录，不太清楚为什么这样设计。

**Comment by lisiyizu** [comment]
建议加上类似语雀的目录大纲

**Comment by mikaelzero** [comment]
+1  加个目录

**Comment by chunyang-wen** [comment]
> 我也赞成目录树的结构，当前一级目录很不方便。
> 楼上举例的‘工作’文件夹很贴近使用场景。
> 不过貌似很多笔记类软件都是一级目录，不太清楚为什么这样设计。

例如 bear，是用 tag 管理的好像。tag 也可以有层级。

**Comment by whsheng** [comment]
针对一些需要显示层级文件，建议增加目录树或tag方式。

**Comment by bz404** [comment]
赞同,建议考虑增加层级目录,树形,挺好:)

**Comment by insureyang** [comment]
只有1级目录但没有目录树确实不方便归档

**Comment by eli-yip** [comment]
希望添加对于单一文本的标题目录，不然对于长文件的编辑真的太折磨了
